drivers/hmis/app/graphql/mutations/submit_form.rb

```diff
@@ -96,7 +96,26 @@ def _resolve(input:, record_lock_version: nil)
           HmisExternalApis::AcHmis::CreateReferralRequestJob.perform_now(record)
         when Hmis::Hud::Enrollment
           record.client.save! if record.client.changed? # Enrollment form may create or update client
-          if record.new_record? || record.in_progress?
+
+          config = Hmis::ProjectAutoEnterConfig.config_for_project(record.project)
+          if config
```

Comment:
We only want to run this for _new_ enrollments.

```suggestion
          # If this is a brand new Enrollment record, and the project has auto-enter configured, save the enrollment as non-WIP and generate an empty intake assessment
          if record.new_record? && Hmis::ProjectAutoEnterConfig.config_for_project(record.project)
```


This mutation is tricky because it's used for lots of different form submissions. If you go to the Enrollment Dashboard and find "Move In Date" or "Date of Engagement" on the overview card, hit edit and it will bring up a modal. That mini-form does a SubmitForm for the Enrollment, in order to change those values. That would be a good thing to QA–– on both WIP and non-WIP enrollments–– to make sure this doesn't break that behavior.

Reply:
Sounds good and thanks for the catch! I tested this locally for both wip and non-wip cases, move in date and engagement date both still seem to behave normally. I also added a note to the PT ticket that we should check this behavior in the QA environment as well.